airflow-core/src/airflow/exceptions.py

```diff
@@ -250,6 +250,12 @@ class DagIsPaused(AirflowException):
     """Raise when a dag is paused and something tries to run it."""
 
     def __init__(self, dag_id: str) -> None:
+        warnings.warn(
```

Comment:
Just remove it -- it isn't in any released Airflow core version.

Reply:
Let me do it in the other PR in that case. No need for two of them.

Reply:
Closing in favour of https://apache-airflow.slack.com/archives/C07813CNKA8/p1745061399226959

Reply:
Actually...this will be released as is with airflow3 right? If that's the case we will need to depreacte it unless we patch this code?